### PR TITLE
BC-266 load baseUrl from api call

### PR DESCRIPTION
**Pull request by pkleybolte**
# Short Description

Get API_URL via http call, this prepares us if there ever will be nuxt without the middleware and also makes this configuration more transparent. This also eliminates the necessity of inline scripting to get that env var

## Links to Ticket and related Pull-Requests
https://ticketsystem.hpi-schul-cloud.org/browse/BC-266

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Review by Loki-Afro — DISMISSED** [review]

**Review by Loki-Afro — APPROVED** [review]

File: src/plugins/axios.js [hunk 1]
```diff
@@ -2,7 +2,13 @@ import AuthModule from "@/store/auth";
 
 const unrecoverableErrorCodes = [401, 404, 500];
 
-export default function ({ $axios, store, error }) {
+export default async function ({ $axios, store, error }) {
+	const runtimeConfigJson = await $axios.get(
+		`${window.location.origin}/runtime.config.json`
+	);
```
Comment (virgilchiriac) [inline src/plugins/axios.js:8]: is it not possible to request the json directly, rather than making a ajax call?

File: src/plugins/axios.js [hunk 2]
```diff
@@ -2,7 +2,13 @@ import AuthModule from "@/store/auth";
...
+	window.schoolCloudRuntimeConfig = runtimeConfigJson.data;
+	$axios.setBaseURL(window.schoolCloudRuntimeConfig.apiURL);
+
```
Comment (virgilchiriac) [inline src/plugins/axios.js:11]: I wonder if the baseUrl can be set in a simpler, dynamic way, in which API_URL env would be available
https://axios.nuxtjs.org/helpers/#setbaseurl

have you tried to create a new axios instance? https://axios.nuxtjs.org/extend

File: src/plugins/axios.js (hunk 1 above)
Comment (Loki-Afro) [reply src/plugins/axios.js:8]: how direct do you wanna request something if not like that? there needs to be an http call.

File: src/plugins/axios.js (hunk 2 above)
Comment (Loki-Afro) [reply src/plugins/axios.js:11]: this is the most simple solution possible, we like simplicity.

**Pull request merged**